openshift_resources/oadp/data_protection_application.py

```diff
@@ -1,17 +1,516 @@
+    )
+    default: Optional[bool] = Field(
+        None,
+        description='default indicates this location is the default backup storage location.',
+    )
+
+
+class AccessMode(Enum):
+    ReadOnly = 'ReadOnly'
+    ReadWrite = 'ReadWrite'
+
+
+class VeleroCredential(BaseModel):
+    key: str = Field(
+        ...,
+        description='The key of the secret to select from.  Must be a valid secret key.',
+    )
+    name: Optional[str] = Field(
+        None,
+        description='Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names TODO: Add other useful fields. apiVersion, kind, uid?',
+    )
+    optional: Optional[bool] = Field(
+        None, description='Specify whether the Secret or its key must be defined'
+    )
+
+
+class ObjectStorage(BaseModel):
+    bucket: str = Field(
+        ..., description='Bucket is the bucket to use for object storage.'
+    )
+    caCert: Optional[str] = Field(
+        None,
+        description='CACert defines a CA bundle to use when verifying TLS connections to the provider.',
+    )
+    prefix: Optional[str] = Field(
+        None,
+        description='Prefix is the path inside a bucket to use for Velero storage. Optional.',
+    )
+
+
+class BackupLocationVelero(BaseModel):
+    accessMode: Optional[AccessMode] = Field(
+        None,
+        description='AccessMode defines the permissions for the backup storage location.',
+    )
+    backupSyncPeriod: Optional[Any] = Field(
+        None,
+        description='BackupSyncPeriod defines how frequently to sync backup API objects from object storage. A value of 0 disables sync.',
+    )
+    config: Optional[Dict[str, str]] = Field(
+        None, description='Config is for provider-specific configuration fields.'
+    )
+    credential: Optional[VeleroCredential] = Field(
+        None,
+        description='Credential contains the credential information intended to be used with this location',
+    )
+    default: Optional[bool] = Field(
+        None,
+        description='Default indicates this location is the default backup storage location.',
+    )
+    objectStorage: ObjectStorage = Field(
+        ...,
+        description="ObjectStorageLocation specifies the settings necessary to connect to a provider's object storage.",
+    )
+    provider: str = Field(
+        ..., description='Provider is the provider of the backup storage.'
+    )
+    validationFrequency: Optional[Any] = Field(
+        None,
+        description='ValidationFrequency defines how frequently to validate the corresponding object storage. A value of 0 disables validation.',
+    )
+
+
+class BackupLocation(BaseModel):
+    bucket: Optional[Bucket] = None
+    velero: Optional[BackupLocationVelero] = Field(
+        None,
+        description='BackupStorageLocationSpec defines the desired state of a Velero BackupStorageLocation',
+    )
+
+
+class PodConfigToleration(BaseModel):
```

Comment:
Toleration is from core.v1 package, should be removed from here and import from there instead:
https://github.com/oadp-qe/openshift-resources/blob/main/src/resources/io/k8s/api/core/v1.py#L1584